### PR TITLE
Predicted gas pumps

### PR DESCRIPTION
**Pull request by metalgearsloth**
I wanted to try out atmos and first thing I found.

The original code is oh so very cursed.

Changed it slightly and made it use a float spinbox too.

image

## Breaking changes
- comp-gas-pump-ui-needs-anchor = ui-needs-anchor
- namespace on the 2 atmos events moved to shared, AtmosDeviceDisabled + Enabled.
- namespace on GasPressurePumpComponent.

:cl:
- fix: Fix gas pumps being capped to atmos max pressure and not their own max pressure.
- tweak: Predict gas pump UI and you no longer need to be in details range to examine it.

### CODE REVIEW

File: Content.Shared/Atmos/Components/GasPressurePumpComponent.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    public bool Enabled = true;
+
+    [DataField("inlet")]
+    public string InletName = "inlet";
+
+    [DataField("outlet")]
+    public string OutletName = "outlet";
+
+    [DataField, AutoNetworkedField]
+    public float TargetPressure = Atmospherics.OneAtmosphere;
```
Comment (Partmedia) [inline Content.Shared/Atmos/Components/GasPressurePumpComponent.cs:18]: Does this leak the target pressure setpoint over the network when the gas pump is out of examine range but still inside PVS? This doesn't seem to me like an important issue, but we should probably be aware of what variables we are sending to the client. Perhaps we don't need the out of range examine check anymore?

File: Content.Shared/Atmos/EntitySystems/SharedGasPressurePumpSystem.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+using Content.Shared.UserInterface;
+
+namespace Content.Shared.Atmos.EntitySystems;
+
+public abstract class SharedGasPressurePumpSystem : EntitySystem
+{
+    [Dependency] private   readonly ISharedAdminLogManager _adminLogger = default!;
+    [Dependency] protected readonly SharedAppearanceSystem Appearance = default!;
+    [Dependency] private   readonly SharedPowerReceiverSystem _receiver = default!;
+    [Dependency] protected readonly SharedUserInterfaceSystem UserInterfaceSystem = default!;
```
Comment (Partmedia) [inline Content.Shared/Atmos/EntitySystems/SharedGasPressurePumpSystem.cs:21]: Huh, when did we start aligning variable modifiers like this?

Comment (metalgearsloth) [reply Content.Shared/Atmos/EntitySystems/SharedGasPressurePumpSystem.cs:21]: I do it but I think it's just a me thing.

**Review by Partmedia — APPROVED** [review]
Looks good to me.

**Pull request merged**